### PR TITLE
Support New displayOnly Confg Option

### PR DESCRIPTION
**Pull request by spencersablan**
### Description

This PR uses the new `displayOnly` config option in the button eligibility logic.

```
paypal.Buttons({
    displayOnly: ["vaultable"]
})
```

### Why are we making these changes? Include references to any related Jira tasks or GitHub Issues
**Jira:** https://paypal.atlassian.net/browse/DTPPCPSDK-825

### Groups who should review (if applicable)

❤️ Thank you!

### CODE REVIEW

**Comment by spencersablan** [comment]
Lint is failing because it doesn't have the new constant I added to paypal-sdk-constants.
image

**Review by wsbrunson — COMMENTED** [review]
We should get a unit test up for `src/funding/funding.js`

File: src/funding/funding.js [hunk 1]
```diff
@@ -30,8 +35,34 @@ type IsFundingEligibleOptions = {|
...
+    const { vendors } = fundingEligibility[source];
+
+    // If any vendors are both eligible & vaultable, card is vaultable
+    for (const vendor in vendors) {
+      if (vendors[vendor].eligible && vendors[vendor].vaultable) {
+        return true;
+      }
+    }
+
+    return false;
```
[the remark this answers is not in the document]
Comment (spencersablan) [reply src/funding/funding.js:56]: Object! Love it! 😄

File: src/funding/funding.test.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+      displayOnly: ["vaultable"],
+      components: COMPONENTS.BUTTONS,
+      ...defaultMockFundingOptions,
+    };
+    const fundingEligible = isFundingEligible(FUNDING.CARD, options);
+
+    expect(fundingEligible).toBe(false);
+  });
+
+  test("card should be eligible if displayOnly includes 'vaultable' and any vendor is vaultable", () => {
```
Comment (wsbrunson) [inline src/funding/funding.test.js:71]: may want to add one more test case
```
amex: {
  eligible: false,
  vaultable: true,
},
```

**Review by wsbrunson — APPROVED** [review]
looking swell sir!

File: src/funding/funding.test.js (hunk 2 above)
Comment (spencersablan) [reply src/funding/funding.test.js:71]: Thanks man! Is this possible though? From my understanding, it can't be vaultable if its not eligible, right?

Comment (spencersablan) [reply src/funding/funding.test.js:71]: Thanks for taking time to review this! I also have to fix some type stuff that's throwing errors with `npm run typecheck`

Comment (wsbrunson) [reply src/funding/funding.test.js:71]: idk you wrote code that handles this lol: https://github.com/paypal/paypal-checkout-components/pull/2252/files#diff-f677f43c72524dd42a4f608dd81477d6a6d9e896205422f6c4aaa5847e9ffeafR49-R52

if you don't think that's possible, i would just remove the `vendor.eligible` check from your `.some` function

Comment (spencersablan) [reply src/funding/funding.test.js:71]: Hahah ohhh.. you're right.  I just looked back at a sample query & card vendors can be vaultable but not eligible 😅 I'll add the test case. 😄

**Pull request merged**